### PR TITLE
Fixes #1615: Adds tests for tcms.testcases.views.EditTestCaseView

### LINKED ISSUE #1615: Automated tests for missing coverage in tcms.testcases.views.EditTestCaseView
Kiwi TCMS needs additional automated test(s) as mentioned in subject.

**Definition of Done**:
- full coverage for the module or class mentioned in the title. This means both positive/negative scenarios where this makes sense or scenarios which traverse different branches of the code where that makes sense.
- test class/test method names are descriptive and reflect the behavior of the subject under test.  A good scheme to follow is `Test<ClassUnderTest>.test_[who][what][expected_results]`
- CI pass
- All code review passed & PR merged

Hints:
- Coverage report is at https://codecov.io/gh/kiwitcms/Kiwi/tree/master/tcms
- Devel getting started docs: https://kiwitcms.readthedocs.io/en/latest/contribution.html#testing
- Django testing documentation: https://docs.djangoproject.com/en/3.0/topics/testing/

*This issue is part of Kiwi TCMS open source bounty program. For more information see the link(s) in [bounty-program milestone](https://github.com/kiwitcms/Kiwi/milestone/4)*

### PR DESCRIPTION
**Pull request by pandafy**
Closes #1615

### CODE REVIEW

**Comment by pandafy** [comment]
@atodorov `black` lint test is failing due to [unrelated issues](https://github.com/kiwitcms/Kiwi/runs/1937219561?check_suite_focus=true#step:5:20). Should I format the file in this PR as a separate commit?

**Comment by atodorov** [comment]
> @atodorov `black` lint test is failing due to [unrelated issues](https://github.com/kiwitcms/Kiwi/runs/1937219561?check_suite_focus=true#step:5:20). Should I format the file in this PR as a separate commit?

Must be a newer version of black then. Fix those issues as well, we can't be merging with failing checks anyway.

**Comment by pandafy** [comment]
@atodorov `black` is modifying unicode strings. Can you please verify whether such change is needed? 

```diff
--- tcms/testcases/tests/test_views.py	2021-02-19 17:32:09.823571 +0000
+++ tcms/testcases/tests/test_views.py	2021-02-19 17:32:14.900261 +0000
@@ -41,17 +41,17 @@
     def setUpClass(cls):
         super().setUpClass()
         cls.field = MultipleEmailField()
 
     def test_to_python(self):
-        value = u"zhangsan@localhost"
+        value = "zhangsan@localhost"
         pyobj = self.field.to_python(value)
         self.assertEqual(pyobj, [value])
 
-        value = u"zhangsan@localhost,,lisi@example.com,"
+        value = "zhangsan@localhost,,lisi@example.com,"
         pyobj = self.field.to_python(value)
-        self.assertEqual(pyobj, [u"zhangsan@localhost", u"lisi@example.com"])
+        self.assertEqual(pyobj, ["zhangsan@localhost", "lisi@example.com"])
 
         for value in ("", None, []):
             pyobj = self.field.to_python(value)
             self.assertEqual(pyobj, [])
```

**Comment by atodorov** [comment]
> @atodorov `black` is modifying unicode strings. Can you please verify whether such change is needed?

If black says it is, then it is.

File: tcms/testcases/tests/test_views.py [hunk 1]
```diff
@@ -331,6 +331,107 @@ def test_edit_a_case(self):
...
+                "email_settings-INITIAL_FORMS": "1",
+                "email_settings-MIN_NUM_FORMS": "0",
+                "email_settings-MAX_NUM_FORMS": "1",
+            }
+        )
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _("Edit TestCase"))
```
Comment (atodorov) [inline tcms/testcases/tests/test_views.py:425]: Should assert that the page we're seeing is editing exactly the same object we think it is.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/testcases/tests/test_views.py (hunk 1 above)
Comment (pandafy) [reply tcms/testcases/tests/test_views.py:425]: @atodorov this test class is to check permissions. Isn't making such assertions out of scope of this class?

Comment (atodorov) [reply tcms/testcases/tests/test_views.py:425]: Is that already checked somewhere else?

Comment (pandafy) [reply tcms/testcases/tests/test_views.py:425]: Wouldn't this test suffice it? 
https://github.com/kiwitcms/Kiwi/blob/5293fa3102336a82956e2fdc776e98ad5a5fb90a/tcms/testcases/tests/test_views.py#L321-L332

Comment (atodorov) [reply tcms/testcases/tests/test_views.py:425]: No. That's a POST which redirects to the page showing the results. You are trying to open the Edit page and need to verify that it is indeed editing the object which you want (whose ID is in the URL).

Comment (pandafy) [reply tcms/testcases/tests/test_views.py:426]: Added such tests here: https://github.com/kiwitcms/Kiwi/pull/2257/files#diff-6ae2ec47dc4f60776d7dd22884bad393555fdf14e8fb6f2e620838bb372170e3R319

Comment (atodorov) [reply tcms/testcases/tests/test_views.py:426]: Nope you haven't. These are all different scenarios and they are asserting on different conditions.

**Review by atodorov — CHANGES_REQUESTED** [review]

**Pull request merged**